src/bootstrap.jl

```diff
@@ -359,7 +359,10 @@ function StatsBase.confint(
     par = filter(collect(propertynames(tbl))) do k
         k = string(k)
         # σ is missing in models without a dispersion parameter
-        if k == "σ" && Missing <: eltype(tbl.σ)
+        Tσ = eltype(tbl.σ)
+        # see https://github.com/JuliaStats/MixedModels.jl/pull/861#discussion_r2323094090
+        # for more info on why this logic is so convulated
+        if k == "σ" && ((Tσ === Any && any(ismissing, tbl.σ)) || Missing <: Tσ)
```

Comment:
If I'm not mistaken, this assumes that `Tσ` will only ever be `Any` if there are missing values present. If `Tσ === Any && !any(ismissing, tbl.σ)` then `Missing <: Tσ` because `Missing <: Any`, but you don't want to return `false` in that case because there are no missing values, which means that the model has a dispersion parameter. Right?

Reply:
If that's right, this should be
```suggestion
        if k == "σ" && ((Tσ === Any && any(ismissing, tbl.σ)) || (Tσ !== Any && Missing <: Tσ))
```